### PR TITLE
Cilium: Add unreachable route for pod IP on deletion option

### PR DESCRIPTION
**Pull request by anthonyhaussman**
When a pod is deleted, the route to its IP is replaced with an unreachable route. When a pod is created, the route is replaced with a route to the pod veth (so if an unreachable existed, it's replaced).

Ref:
 - https://github.com/cilium/cilium/pull/18505

### CODE REVIEW

**Comment by anthonyhaussman** [comment]
@olemarkus, @hakman.
I have tested the `enable-unreachable-routes` option on one of our Kubernetes clusters and it worked like a charm.
It helps applications to detect socket destruction and act accordingly (re-create a socket).

I think that PR can be merged but waiting for your approval. :slightly_smiling_face: :pray:

**Comment by hakman** [comment]
Thanks @anthonyhaussman!
/lgtm
/assign @olemarkus

**Review by olemarkus — CHANGES_REQUESTED** [review]
Looks good.
Can you add a small release note?

**Comment by anthonyhaussman** [comment]
> Looks good. Can you add a small release note?

No problem @olemarkus.
Shall I add it to all compatible versions? (`1.25-NOTES.md`/`1.26-NOTES.md`/`1.27-NOTES.md`) or only the last version?

**Comment by olemarkus** [comment]
We can backport this into 1.26, so suggest you add it to those notes.

File: docs/releases/1.26-NOTES.md [hunk 1]
```diff
@@ -82,6 +82,13 @@ CNIs, use the "cni" networking option instead.
 
 * All legacy addons are deprecated in favor of managed addons, including the [metrics server addon](https://github.com/kubernetes/kops/tree/master/addons/metrics-server) and the [autoscaler addon](https://github.com/kubernetes/kops/tree/master/addons/cluster-autoscaler).
 
+# Full change list since 1.25.0 release
+
+
+## 1.26.0-beta.2 to 1.26.0-beta.3
+
+* Add unreachable route for pod IP on deletion option in Cilium [@anthonyhaussman](https://github.com/anthonyhaussman) [#15050](https://github.com/kubernetes/kops/pull/15050)
```
Comment (olemarkus) [inline docs/releases/1.26-NOTES.md:90]: This bit is autogenerated. You can add a section `Other changes of note` towards the end. See https://kops.sigs.k8s.io/releases/1.23-notes/#other-changes-of-note

Comment (hakman) [reply docs/releases/1.26-NOTES.md:90]: Just in the 1.26 release notes, not 1.27 also.

Comment (anthonyhaussman) [reply docs/releases/1.26-NOTES.md:90]: I correct this :ok_hand:

Comment (anthonyhaussman) [reply docs/releases/1.26-NOTES.md]: I hope it's okay now @olemarkus @hakman :slightly_smiling_face:

File: docs/releases/1.26-NOTES.md [hunk 2]
```diff
@@ -86,3 +86,7 @@ CNIs, use the "cni" networking option instead.
 
 * kOps needs maintainers for Canal, Flannel, Kube-Router, and Spotinst to keep versions up to date and move the integration from experimental to stable.
 If no volunteers step up by the time kOps 1.27 is released, support will be phased out.
+
+# Other changes of note
+
+* Add unreachable route for pod IP on deletion option in Cilium [@anthonyhaussman](https://github.com/anthonyhaussman) [#15050](https://github.com/kubernetes/kops/pull/15050)
```
Comment (hakman) [inline docs/releases/1.26-NOTES.md:92]: Please move to the top, just before breaking changes.
```suggestion
* Added unreachable route for pod IP on deletion option in Cilium.
```

Comment (anthonyhaussman) [reply docs/releases/1.26-NOTES.md]: Done :ok_hand:

**Review by olemarkus — APPROVED** [review]

**Pull request merged**